Scripts/Pointer/BasePointer.cs

```diff
@@ -0,0 +1,298 @@
+﻿namespace VRTK.Core.Pointer
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using VRTK.Core.Utility;
+    using VRTK.Core.Data.Type;
+
+    /// <summary>
+    /// The PointerData contains information about the current pointer state.
+    /// </summary>
+    public class PointerData : SurfaceData
+    {
+        /// <summary>
+        /// Determines if the pointer is currently in it's activated state.
+        /// </summary>
+        public bool isActive;
+        /// <summary>
+        /// Determines if the pointer is in it's valid state.
+        /// </summary>
+        public bool isValid;
+        /// <summary>
+        /// The duration that the pointer has been hovering over it's current target.
+        /// </summary>
+        public float hoverDuration;
+    }
+
+    /// <summary>
+    /// The BasePointer forms the basis for all pointer types.
+    /// </summary>
+    public abstract class BasePointer : MonoBehaviour
+    {
+        /// <summary>
+        /// The visibility of the pointer element.
+        /// </summary>
+        public enum VisibilityType
+        {
+            /// <summary>
+            /// The element will only be visible when the pointer is in the active state.
+            /// </summary>
+            OnWhenActive,
+            /// <summary>
+            /// The element will always be visible regardless of the pointer state.
+            /// </summary>
+            AlwaysOn,
+            /// <summary>
+            /// The element will never be visible regardless of the pointer state.
+            /// </summary>
+            AlwaysOff
+        }
+
+        [Header("Base Settings")]
+
+        [Tooltip("An optional ExclusionRule to determine valid and invalid targets based on the set rules.")]
+        public ExclusionRule exclusionRule;

```

Comment:
What about renaming this to `validityExclusionRule` to allow classes that inherit from this abstract class to have additional exclusion rules without confusing users because of the naming?

Reply:
`ExclusionRule` isn't an abstract class it's basically the new name for `PolicyList` because it's basically a `Policy` but the word list was dumb

and `PolicyRule` is also dumb because they mean the same thing.

Maybe `ExclusionRule` should be simply called `Policy` and you just check if the policy should exclude the thing you're checking on `Policy.ShouldExclude(thing)`

then in this case you could have the variable called `validityPolicy`

Reply:
I think `ExclusionRule` is a good name, though I like `Policy`, too. 🤔

For this comment I only am interested in changing the variable name. I can see subclasses perhaps needing a different "rule" for other functionality and having a generically named `exclusionRule` on the superclass makes that harder to understand.

`exclusionRule`'s documentation already states it's only used "to determine valid and invalid targets".